src/server/services/generate-arpa-report.js

```diff
@@ -12,6 +11,33 @@ function isNotNull (value) {
   return value != null
 }
 
+function isProjectRecord (record) {
+  return [
+    'ec 1 - public health',
+    'ec 2 - negative economic impact',
+    'EC 3 - Public Sector Capacity',
+    'EC 4 - Premium Pay',
+    'EC 5 - Infrastructure',
+    'EC 7 - Admin'
+  ].includes(record.type)
+}
+
+/**
+ * Extract the Detailed Expenditure Category code from a record.
+ *
+ * @returns {string} The detailed EC code in format "#.##".
+ */
+function getDetailedEcCode (record) {
+  const EC_CODE_REGEX = /^(\d.\d\d?)/
+  const { subcategory } = record
+
+  if (EC_CODE_REGEX.test(subcategory)) {
+    const [, detailedECCode] = EC_CODE_REGEX.exec(subcategory)
+    return detailedECCode
+  }
+  return undefined
```

Comment:
```suggestion
  const match = EC_CODE_REGEX.exec(subcategory)
  return match ? match[1] : undefined
```

avoids having to run the regex twice, plus seems a little easier to read

Reply:
👌   I considered a few approaches to this, and now I know which you prefer!